app/lib/features/space/settings/pages/apps_settings_page.dart

```diff
@@ -441,6 +344,181 @@ class SpaceAppsSettingsPage extends ConsumerWidget {
+    int maxPowerLevel,
+    int? currentPw,
+    Space space,
+    RoomPowerLevels powerLevels,
+  ) async {
+    final newPowerLevel = await showDialog<int?>(
+      context: context,
+      builder: (BuildContext context) => ChangePowerLevel(
+        featureName: 'Pin',
+        currentPowerLevelName:
+            maxPowerLevel == 100 ? powerLevelName(currentPw) : 'Custom',
+        currentPowerLevel: currentPw,
+      ),
+    );
+    if (newPowerLevel == currentPw) return;
+    await space.updateFeaturePowerLevels(powerLevels.pinsKey(), newPowerLevel);
+    if (context.mounted) {
+      EasyLoading.showToast(L10n.of(context).powerLevelSubmitted('Pins'));
+    }
+  }
+
+  Future<void> onPinToggle(
+    PinsSettings pins,
+    bool newVal,
+    ActerAppSettings appSettings,
+    Space space,
+  ) async {
+    final updated = pins.updater();
+    updated.active(newVal);
+    final builder = appSettings.updateBuilder();
+    builder.pins(updated.build());
+    await space.updateAppSettings(builder);
+  }
+
+  Future<void> onCalendarEventLevelChange(
+    BuildContext context,
+    int maxPowerLevel,
+    int? currentPw,
+    Space space,
+    RoomPowerLevels powerLevels,
+  ) async {
+    final newPowerLevel = await showDialog<int?>(
+      context: context,
+      builder: (BuildContext context) => ChangePowerLevel(
+        featureName: 'Calendar Events',
+        currentPowerLevelName:
+            maxPowerLevel == 100 ? powerLevelName(currentPw) : 'Custom',
+        currentPowerLevel: currentPw,
+      ),
+    );
+    if (newPowerLevel == currentPw) return;
+    await space.updateFeaturePowerLevels(
+      powerLevels.eventsKey(),
+      newPowerLevel,
+    );
+    if (context.mounted) {
+      EasyLoading.showToast(
+        L10n.of(context).powerLevelSubmitted('Calendar Events'),
+      );
+    }
+  }
+
+  Future<void> onCalendarEventToggle(
+    EventsSettings events,
+    bool newVal,
+    ActerAppSettings appSettings,
+    Space space,
+  ) async {
+    final updated = events.updater();
+    updated.active(newVal);
+    final builder = appSettings.updateBuilder();
+    builder.events(updated.build());
+    await space.updateAppSettings(builder);
+  }
+
+  Future<void> onTaskListLevelChange(
+    BuildContext context,
+    int maxPowerLevel,
+    int? currentPw,
+    Space space,
+    RoomPowerLevels powerLevels,
+  ) async {
+    final newPowerLevel = await showDialog<int?>(
+      context: context,
+      builder: (BuildContext context) => ChangePowerLevel(
+        featureName: 'Task Lists',
+        currentPowerLevelName:
+            maxPowerLevel == 100 ? powerLevelName(currentPw) : 'Custom',
+        currentPowerLevel: currentPw,
+      ),
+    );
+    if (newPowerLevel == currentPw) return;
+    await space.updateFeaturePowerLevels(
+      powerLevels.taskListsKey(),
+      newPowerLevel,
+    );
+    if (context.mounted) {
+      EasyLoading.showToast(L10n.of(context).powerLevelSubmitted('Task Lists'));
+    }
+  }
```

Comment:
these all seem rather similar ... like, we could easily make one method that combines them and only paste in the relevant featureName, specificKey and be good ...

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1580/commits/531ba2acb31557ede1b8a2f8d2a0017cde2e58de